keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -515,25 +521,63 @@ Similar to the BucketAccess for COSI created bucket, this BucketAccess should re
 
-When a user deletes a BucketAccess, the corresponding secret/serviceaccount are also deleted. If a pod has that secret mounted when delete is called, then a finalizer on the secret will prevent it from being deleted. Instead, the deletionTimestamp will be set on the secret. In this way, access to a Bucket is preserved until the application pod dies. 
+```yaml
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: bucket-and-access-claim-quota
+spec:
+  hard:
+    count/bucketclaims.objectstorage.k8s.io: "1"
+    count/bucketaccesses.objectstorage.k8s.io: "10"
+```
+
+### Internal Quotas
+
+Quotas internal to the object store vary between object storage vendors. For example, many on-premises object storage providers, like Ceph RGW and Minio, allow administrators to configure limits on the size or quantity of objects either per-bucket or per-user (per-access). Cloud providers, like AWS and GCP, do not provide similar quota options.
+
+One of COSI's primary goals is portability, and optionally-implemented top-level fields may negatively impact portability. As such, COSI does not plan to add specifications for non-standardized quotas on a per-bucket or per-user (per-access) basis.
+
+COSI instead recommends that vendor drivers utilize the opaque `Parameters` field available in BucketClass and BucketAccessClass (and Buckets, for brownfield use-cases) to allow setting vendor-specific quotas. Vendor-specific parameters are allowed on administrator-configured resources because these do not constitute the user-facing, portable API specification.
+
+#### User self-service quotas
 
-When an admin deletes any of the class objects, it does not affect existing Buckets as fields from the class objects are copied into the Buckets during creation. 
+Administrators have reported a desire to provide users with the ability to self-select their own quotas, sometimes within certain bounds. The issue that administrators have taken with COSI is that it does not provide an opaque field on user-facing, portable APIs (BucketClaim and BucketAccess). This is deliberate on COSI's part, to ensure portability, which would be broken by non-standardized, vendor-specific fields. COSI's does not intend to add API-level support for these use-cases.
 
-If a Bucket is manually deleted by an admin, then a finalizer on the Bucket prevents it from being deleted until the binding BucketClaim is deleted. 
+Below are some examples of how to achieve the desired outcome within the bounds of what COSI is able to provide. Be mindful that implementing these options will add difficulty (sometimes substantial) when porting applications between different cloud environments.
+
+* The administrator may create an array of Bucket(Access)Classes and allow users to self-select the closest provided option that matches their needs.
+* If that is too limiting, administrators may create a web form or custom controller that can create a custom Bucket(Access)Class for a user on-demand.
```

Comment:
Take a look at the VolumeAttributesClass KEP: https://github.com/kubernetes/enhancements/pull/3780

Reply:
https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/3751-volume-attributes-class/README.md#appendix---current-sps-case-study